% Aligning of sequences can be posed as a mathematical problem. Consider two sequences $S_1$ and $S_2$ (possibly of different lengths) with residues from some finite alphabet $D$. Let one of the sequences (usually the longer one) be the \textit{reference sequence}; this is the sequence that the other one will be aligned to. The typical alignment approach considers the edit distance between the two sequences. This is achieved by either inserting, deleting or substituting characters in certain positions in the sequence to be aligned and assigning penalties to these operations, while scoring matching positions. An alignment is obtained through using a scoring scheme and optimizing an objective function. 

Aligning of sequences can be posed as a mathematical problem. Consider two sequences $S_1$ and $S_2$ (possibly of different lengths) with residues from some finite alphabet $D$. Let one of the sequences (usually the longer one) be the \textit{reference sequence}; this is the sequence that the other one will be aligned to.
A standard implementation of pairwise sequence alignment is obtained by considering the edit distance between two sequences and optimising a scoring function. The edit distance is obtained by considering the number of insertions, deletions and substitutions of residues in the sequence to be aligned relative to the reference sequence. Figure~\ref{fig:matrix} shows a sample scoring matrix while Table~\ref{tab:alignment} shows an example of an alignment obtained with the aforementioned scoring scheme. To achieve an optimal pairwise alignment the two sequences are stored in a table with the residues in columns and a (possibly negative) score is assigned to each column. %Typically, columns where insertions, deletions or substitutions have taken place incur a penalty, while matching sequence entries have an associated positive score. 
The scoring system plays a large role in the generation of the alignment and there are widely accepted scoring schemes such as PAM or BLOSSUM~\cite{mount2008comparison}. The selection of these scoring matrices typically depends of the application. 

\begin{figure}[H]
    \centering
    \begin{blockarray}{cccccc}
    & A & C & G & T & -\\
\begin{block}{c(ccccc)}
  A & 5 & 1 & 1 & 2 & -2 \\
  C & 1 & 5 & 2 & 1 & -2 \\
  G & 1 & 2 & 5 & 1 & -2 \\
  T & 2 & 1 & 1 & 5 & -2 \\
  - & -2 & -2 & -2 & -2 & -10 \\
\end{block}
\end{blockarray}
    \caption{A sample scoring scheme. Matching bases are given large positive scores. Mismatches are awarded lower scores, while inserted gaps, denoted by the ``-" character incur a penalty.}
    \label{fig:matrix}
\end{figure}
% \begin{table}[]
% \begin{tabular}{clc}
% \multicolumn{1}{l|}{Unaligned Sequences} & \multicolumn{1}{l|}{Scoring Scheme}                                         & \multicolumn{1}{l}{Aligned Sequences} \\ \hline
% $S_1=\texttt{TTCGAT}$                    & \multirow{2}{*}{\begin{tabular}[c]{@{}l@{}}$\begin{blockarray}{cccccc}
% a & b & c & d & e \\
% \begin{block}{(ccccc)c}
%   1 & 1 & 1 & 1 & 1 & f \\
%   0 & 1 & 0 & 0 & 1 & g \\
%   0 & 0 & 1 & 0 & 1 & h \\
%   0 & 0 & 0 & 1 & 1 & i \\
%   0 & 0 & 0 & 0 & 1 & j \\
% \end{block}
% \end{blockarray}$\end{tabular}} & $S_1=\texttt{TTCGAT}$                 \\
% $S_2=\texttt{TGCAT}$                     &                                                                             & $S_2=\texttt{T-GCAT}$                
% \end{tabular}
% \end{table}

   
\begin{minipage}{\linewidth}
\centering
\captionof{table}{An Example Alignment} \label{tab:alignment}
\begin{tabular}{clc}
\multicolumn{1}{|l|}{Unaligned Sequences} & \multicolumn{1}{l|}{\multirow{3}{*}{$\rightarrow$}} & \multicolumn{1}{l|}{Aligned Sequences} \\ \cline{1-1} \cline{3-3} 
$S_1=\texttt{TTCGAT}$                     & \multicolumn{1}{l}{}                               & $S_1=\texttt{TTCGAT}$                  \\
$S_2=\texttt{TGCAT}$                      & \multicolumn{1}{l}{}                               & $S_2=\texttt{T-GCAT}$                 
\end{tabular}\par
{An example alignment obtained by using the scoring scheme from Figure~\ref{fig:matrix}. The resulting alignment is obtained by inserting a gap between the first and second nucleotides of $S_2$. }
\end{minipage}
    % \label{fig:alignment}

A variety of algorithms exist for the purpose of multiple sequence alignment (MSA) as opposed to just a pair. The class of algorithms associated with alignment problems, lends itself to the style of dynamic programming. Finding a pairwise alignment of two sequences is typically obtained by inserting gaps into the shorter sequence that maximizes the similarities of the input sequences according to some scoring matrix~\cite{edgar2006multiple}. One of the standard approaches for constructing multiple sequence alignments for $N$ sequences is to do $N-1$ pairwise alignments of pairs of sequences, under guidance from phylogenetic trees~\cite{feng1987progressive}.
	The alignment of nucleotide or protein sequences is useful because it can give insight into the different mutations that have occurred through genetics that give rise to different phenotypes. 


The process described above it known as global sequence alignment. By attempting to optimize the overall alignment of the two sequences, long regions of low similarity may be included~\cite{needleman1970general}. In contrast, local alignments seek similarity between subsequences that are somewhat conserved; in this approach unconserved regions not  contributeto the similarity measure~\cite{goad1982pattern, sellers1984pattern, smith1981comparison}. As with global sequence alignment, there exist scoring matrices; the PAM and BLOSSUM matrices have variants for local alignment~\cite{mount2008comparison}. 
% 	\subsection{Genetic Segmentation}
% 	The act of segmenting a genetic sequence according to some criterion is known as genetic segmentation. There is some debate as to whether or not the genome actually follows a segmented structure {\color{red}CITE}. 
% 	% 
% PAM OR BLOSUM \cite{mount2008comparison}