%A typical alignment based approach considers the edit distance between the two sequences. This is achieved by either inserting, deleting or substituting characters in certain positions in the sequence to be aligned and assigning penalties to these operations, while scoring matching positions. An alignment is obtained through using a scoring scheme and optimizing an objective function. 
           % {\color{red}Disclaimer: This might not actually be how it works.}
            %A standard implementation of pairwise sequence alignment is obtained by considering the edit distance between two sequences and optimizing a scoring function. The edit distance is obtained by considering the number of insertions, deletions and substitutions of residues in the sequence to be aligned relative to the reference sequence. To achieve an optimal pairwise alignment the two sequences are stored in a table with the residues in columns and a (possibly negative) score is assigned to each column. Typically, columns where insertions, deletions or substitutions have taken place incur a penalty, while matching sequence entries have an associated positive score. The scoring system plays a large role in the generation of the alignment. As such, at least in the field of bioinformatics there are widely accepted scoring schemes such as PAM or BLOSSUM~\cite{mount2008comparison}. The selection of these scoring matrices typically depends of the application.
            
            
        \subsection{Alignment based approaches}
            One of the main tools in comparative genomics is the sequence alignment. 
            Consider two sequences $S_1$ and $S_2$ (possibly of different lengths) with residues from some finite alphabet $D$. Let one of the sequences (usually the longer one) be the \emph{reference sequence}; this is the sequence that the other one will be aligned to. 
            %A typical alignment approach considers the edit distance between the two sequences. This is achieved by either inserting, deleting or substituting characters in certain positions in the sequence to be aligned and assigning penalties to these operations, while scoring matching positions. An alignment is obtained through using a scoring scheme and optimizing an objective function. 
            {\color{red}Disclaimer: This might not actually be how it works.}
            A pairwise sequence alignment can be obtained by considering the edit distance between two sequences and optimizing a scoring function. The edit distance is obtained by considering the number of insertions, deletions and substitutions of residues in the sequence to be aligned relative to the reference sequence. To achieve an optimal pairwise alignment the two sequences are stored in a table with the residues in columns and a (possibly negative) score is assigned to each column. Typically, columns where insertions, deletions or substitutions have taken place incur a penalty, while matching sequence entries have an associated positive score. The scoring system plays a large role in the generation of the alignment. In the field of bioinformatics the sequence edits are interpreted as point mutations between the two species. This has lead to the development of PAM~CITEPAM or BLOSSUM~CITEBLOSSUM. The selection of these scoring matrices typically depends of the application~\cite{mount2008comparison}.

            A variety of algorithms exist for the purpose of multiple sequence alignment (MSA) as opposed to just a pair of them. The class of algorithms associated with alignment problems, lends itself to the style of dynamic programming. Finding a pairwise alignment of two sequences is typically obtained by inserting gaps into the shorter sequence that maximizes the similarities of the input sequences according to some scoring matrix~\cite{edgar2006multiple}. One of the standard approaches for constructing multiple sequence alignments for $N$ sequences is to do $N-1$ pairwise alignments of pairs of sequences, under guidance from phylogenetic trees~\cite{feng1987progressive}.
        % \subsection{This is the method that we use}
        %     It is grounded in alignment based methods.
        %     Sometimes there are deterministic ones but we focus on stochastic methods
        % \subsection{Here are other methods that we can try}