Use the methylation data to infer the chromatin states of each organism. Combine the DNA sequence here with the methylation data. What sort of prior would one place on this? How does one choose to map some biological explanation with some observed phenomenon. Well, we are able to say that these particular positions are methylated. Perhaps when you have some combination of methylation marks (we onlt have one) we can say whether we can use this to produce a profile of the chromatin state. Among the sequences located in regions that are inferred to be in an open chromatition, we form a library for each organism. We identify motifs that are candidates for TFBSs. Alignment free sequenc comparison can be maybe used to see what motifs are conserved. 

The methylation data can act as a filter for sequences in the open state and this is important because this profile will vary according to the cell type and function and to a degree, species. We have a focus on cardiac cells at various stages of development. These differing profiles can inform us on the enhancer regions that are conserved across species. 

See if there exists some pre-existing work that models chromatin state based on methylation data in a bayesian approach. Then you can judge model assumptions, select the ones that you agree with and reasses the ones that you are less comfortable with. 

Hopefully at the bare minimum something like that exists. If it doesn't, see what anyone has done with ChIP-Seq data in a bayesian setting. Maybe even start comparing some frequentist approaches and see what comparisons with bayesian inferrences and how this may translate over to our context. If no ChIP-Seq methods have been used, compare with previous generation sequencing think about how these inferences are robust when considering the difference getting to comparable ChIP-Seq stuff. 