\documentclass[12pt,a4paper]{article}
\author{D Maderazo}
\usepackage{amsmath, amsthm, amssymb, textcomp, enumerate, multicol, fancyhdr, varwidth, graphicx, color, mathrsfs, tikz, pgfgantt, epstopdf}
\usepackage[left=25mm,right=25mm,top=30mm,bottom=30mm]{geometry}
\usepackage{amsfonts, float}
\usepackage[]{algorithm2e}
\usepackage{csquotes}

\usepackage{xcolor,colortbl}

\linespread{1}
\newtheorem*{mydef}{Definition}
\newtheorem{mythm}{Theorem}
\pagestyle{fancy}
\lhead{}
\rhead{Dominic Maderazo}
\newcommand*\dee{\mathop{}\!\mathrm{d}}
\newcommand*\del{\mathop{}\!\partial}
\def\*#1{\mathbf{#1}}
\setlength{\parskip}{1em}
\usetikzlibrary{shapes.geometric, arrows}
\tikzstyle{startstop} = [rectangle, rounded corners, minimum width=3cm, minimum height=1cm,text centered, draw=black, fill=red!30]
\tikzstyle{io} = [trapezium, trapezium left angle=70, trapezium right angle=110, minimum width=3cm, minimum height=1cm, text centered, draw=black, fill=blue!30]
\tikzstyle{process} = [rectangle, minimum width=3cm, minimum height=1cm, text centered, draw=black, fill=orange!30]
\tikzstyle{decision} = [diamond, minimum width=3cm, minimum height=1cm, text centered, draw=black, fill=green!30]
\tikzstyle{arrow} = [thick,->,>=stealth]

\usepackage{forloop}
\newcounter{loopcntr}
\newcommand{\rpt}[2][1]{%
  \forloop{loopcntr}{0}{\value{loopcntr}<#1}{#2}%
}
\newcommand{\on}[1][1]{
  \forloop{loopcntr}{0}{\value{loopcntr}<#1}{&\cellcolor{gray}}
}
\newcommand{\off}[1][1]{
  \forloop{loopcntr}{0}{\value{loopcntr}<#1}{&}
}


% \linespread{2}
% \setlength{\parindent}{4em}        
\begin{document}
% \title{Application of Bayesian Techniques for Genome Segmentation Analysis}
% \date{\today}
% \author{Dominic Maderazo}
% \begin{center}
%     {Supervisors: Jennifer Flegg \& Jonathan Keith}
% \end{center}%Supervisors: Jennifer Flegg \& Jonathan Keith}
% \renewcommand{\maketitlehookb}{\centering You won't expect the results}
% \maketitle
\begin{titlepage}
	\centering
	\includegraphics[width=0.15\textwidth]{PRIMARY_A_Vertical_Housed_CMYK.eps}\par\vspace{1cm}
	{\scshape\LARGE The University of Melbourne \par}
	\vspace{1cm}
	{\scshape\Large Confirmation Report\par}
	\vspace{1.5cm}
	{\huge\bfseries Computational Methods for Regulatory Enhancer Identification\par}
	\vspace{2cm}
	{\Large\itshape Dominic Maderazo\par}
	\vfill
	supervised by\par
	Dr.~Jennifer \textsc{Flegg}, Dr.~James \textsc{Osrborne} \& Assoc.~Prof.~Jonathan \textsc{Keith}

	\vfill

% Bottom of the page
	{\large \today\par}
\end{titlepage}
    % \section*{Abstract}
    % Yes
    % sub Sections for lit review: (levels *, and > )
    %     * Bio background
    %     * Enhancer Identification
    %     * Mathematics
    %         > Param Est
    %         > Markov Chains
    %         > HMMs
    %         > Alignment
    %         > Change Point
    %         > Heirarchical modelling?
    %         > MEME
    

    \section{Literature Review}
        \subsection{Biological Background}
        \input{Enhancers}
        \subsection{Computational Identification of Enhancers}
        \label{sSec:computational}
        \input{compIdent} 
        \subsection{Cardiac Enhancers}
        \input{HeartStuff}
        \subsection{Mathematical Techniques}
        \label{sSec:mathematical}
        A focus will be placed on those techniques that are comparative in nature and those that are able to take into account epigenetic markers as input. After an introduction Markov chains and paramater estimation necessary for a more detailed discussion of the mathematics underlying the computational methods discussed in Section~\ref{sSec:mathematical}. In this section there will be a discussion of HMMs, alignment based and alignment free sequence comparison. These methods are chosen because of how they are able to ChromHMM is a great example of incorporating epigenetic sequence information, while alignment based and alignment free methods re able to use ideologies of comparative genomics effectively. It is important to mention the MEME family of algorithms as the method is prevalent in the field of TFBS identification, but they will not be discussed here.%They will not be discussed here as they do not take into account available epigenetic information, nor are they comparative. 
        Interested readers are directed to the MEME suite website (\texttt{http://meme-suite.org/}) and Section~8.5-6 of Parida~\cite{parida2007pattern}. 

            % \subsubsection{k-Means Clustering}
            % \input{kMeans}
            

            \subsubsection{Markov Chains}
            \input{markovChains}
            
            \subsubsection{Parameter Estimation}
            \input{paramEst}
            
            \subsubsection{Markov Chain Monte Carlo}
            \input{MCMC}
            
            % \subsubsection{MEME}
            % \input{MEMEs} 
            \subsubsection{Hidden Markov Models}
            \input{HMMs} 
            
            \subsubsection{Sequence Alignment}
            \input{alignment}
            
            \subsubsection{Change Point Detection}
            \input{Segmentation}
            
            
            \subsubsection{Alignment Free Sequence Comparison}
            \input{alignmentFree}
            
            
            
            
    % \section{Conceptual Framework}
        
    % \section{Preliminary Data}
    \section{Relevance of study}
        There is some mystery surrounding cardiac enhancers as they are not conserved in the same way that enhancers in other tissue may be, yet are still able to retain function. Being able to rank the importance of enhancers has been done in other tissue in the past by using conservation as a metric, but a lack of conservation for cardiac enhancers poses a problem.
        The aim of the project is to extend existing methods for the computational identification of conserved developmental cardiac enhancers or the development of new ones that are able to rank the importance of enhancers. The the extent that the literature was reviewed, there appears to be a gap in computational methods that function this way in cardiac specific tissue.
        
        We have access to ChIP-Seq H3K4me1 data collected by collaborators for different species in heart tissue in conjunction with those that are available in public databases, at a variety of developmental stages. This type of methylation has been used to identify active and primed enhancers, but it is poorly understood whether H3K4me1 influences, is influenced by or correlates enhancer activity~\cite{rada2018h3k4me1}. This methylation of this particular site often associated with promoter regions of actively being transcribed genes~\cite{barski2007high}. We would like to develop methods that are able to use this data for the computational identification of heart enhancers or extend existing methods to include this data. Whatever method gets developed, it is aimed to be published in the public domain to be used as a tool for bioinformatics and applicable in different tissues specific contexts.  
        
    \section{Progress to date}
    \input{progress}
    
    \section{Work to be done}
    % \begin{figure}[H]
    %     \centering
    %     \input{gantt}
    %     \caption{Caption}
    %     \label{fig:my_label}
    % \end{figure}
    
    \input{futureStuff}
    
    \section{Timeline}
\definecolor{barblue}{RGB}{153,204,254}
\definecolor{groupblue}{RGB}{51,102,254}
\definecolor{linkred}{RGB}{165,0,33}


\begin{ganttchart}[
   canvas/.append style={fill=none, draw=black!5, line width=.75pt},
   hgrid style/.style={draw=black!5, line width=.75pt},
   vgrid,%={*1{draw=black!15, line width=.75pt}},
   today=0,
   today rule/.style={
     draw=black!64,
     dash pattern=on 3.5pt off 4.5pt,
     line width=1.5pt
   },
   today label font=,%\small\bfseries,
   title/.style={draw=none, fill=none},
   title label font=\bfseries,
   title label node/.append style={below=7pt},
   include title in canvas=false,
   % bar label font=\mdseries\small\color{black!70},
%     bar label node/.append style={align=left,text width=14em,below left=-5pt and 0pt},
   bar/.append style={fill=groupblue,draw=none},
   bar incomplete/.append style={fill=barblue},
   % bar progress label font=\mdseries\footnotesize\color{black!70},
   group incomplete/.append style={fill=groupblue},
   group left shift=0,
   group right shift=0,
   group height=.5,
   group peaks tip position=0,
   group label node/.append style={align=left,text width=15em},
        Mile2/.style={milestone/.append style={fill=yellow,shape=circle}}
   % group progress label font=\bfseries\small,
 ]{1}{12}
    \sffamily
 \gantttitle[
   title label node/.append style={below left=7pt and -3pt}
 ]{Months from now:\quad}{0}
 \gantttitlelist{0,2,...,24}{1} \\
    \ganttbar{Development of \texttt{R} library}{1}{3}
    \ganttmilestone[Mile2]{}{5}\\ 
%  \ganttbar{Model of nuclear calcium}{1}{2} \\
%     \ganttmilestone[Mile2]{}{9}
 \ganttbar{Alignment Based Enhancer Identification}{2}{5} 
 \ganttmilestone[Mile2]{}{6}
    \\
    % \ganttmilestone[Mile2]{}{10}
    % \ganttbar{Simplified model of channel interaction}{5}{7} \\
    % \ganttmilestone[Mile2]{}{12}
    \ganttbar{Alignment Free Enhancer Identification}{6}{9}
    \ganttmilestone[Mile2]{}{10} \\
    \ganttbar{Thesis write up}{10}{12}
    % \ganttlink{elem2}{elem3}

\end{ganttchart}\\
\caption{The period of work is indicated with blue bars while yellow circles indicate write up of the work for publication.}\\
    \bibliographystyle{acm}
\bibliography{LitReview}
\end{document}

Plan:
    Literature review
        Detail what enhancers are, what are their purpose 
        There's 